lib/qs/event_handler.rb

```diff
@@ -0,0 +1,38 @@
+require 'qs/event'
+require 'qs/job_handler'
+
+module Qs
+
+  module EventHandler
+
+    def self.included(klass)
+      klass.class_eval do
+        include Qs::JobHandler
+        include InstanceMethods
+      end
+    end
+
+    module InstanceMethods
+
+      def initialize(*args)
+        super
+        @qs_event = Event.new(@qs_runner.job)
```

Comment:
@jcredding can't you do `Event.new(self.job)` since we are after the `super` call?